### PR TITLE
[FIX] Only import required components from torch

### PR DESCRIPTION
**Pull request by ravinkohli**
Due to the high memory consumption causing crashes for some of the dataset, this PR aims to solve this issue partly. It is based on @franchuterivera's suggestions detailed in #180. Specifically in this PR, all torch imports have been modified to only import components required in the file

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

Note that a Pull Request should only contain one of refactoring, new features or documentation changes.
Please separate these changes and send us individual PRs for each.
For more information on how to create a good pull request, please refer to [The anatomy of a perfect pull request](https://medium.com/@hugooodias/the-anatomy-of-a-perfect-pull-request-567382bb6067).

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

## Description

Only the required components are imported from all the torch statements throughout the library.

## Motivation and Context

See issue #180. Furthermore, I have used a memory profiler on `examples/20_basics/example_tabular_classification.py` to measure the difference between the current `development` branch and the solution proposed in this PR. I found that the max memory usage through the course of the experiment (measured every 0.1 seconds) is reduced by 400MB and the average memory used is reduced by 77MB. While this does not mean this solution is responsible for the entirety of the reduction but it shows that this change helps reduce our memory footprint.

### CODE REVIEW

**Comment by nabenabe0928** [comment]
It seems the following files still include `import torch`:
./autoPyTorch/datasets/image_dataset.py
./autoPyTorch/datasets/tabular_dataset.py
./autoPyTorch/datasets/time_series_dataset.py
./autoPyTorch/datasets/base_dataset.py
./autoPyTorch/pipeline/base_pipeline.py
./autoPyTorch/pipeline/components/setup/traditional_ml/base_model.py
./autoPyTorch/pipeline/components/setup/early_preprocessor/utils.py
./autoPyTorch/pipeline/components/training/data_loader/feature_data_loader.py
./autoPyTorch/pipeline/components/training/data_loader/image_data_loader.py
./autoPyTorch/pipeline/components/training/trainer/__init__.py
./autoPyTorch/pipeline/components/preprocessing/image_preprocessing/normalise/ImageNormalizer.py
./autoPyTorch/pipeline/components/preprocessing/image_preprocessing/normalise/NoNormalizer.py
./autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/TabularColumnTransformer.py
./autoPyTorch/pipeline/components/preprocessing/base_preprocessing.py

**Comment by nabenabe0928** [comment]
The following two codes consume the same amount of memory.
```python:bench.py
import torch
import time

time.sleep(1)
torch.rand(1)
```

```python:bench.py
from torch import rand
import time

time.sleep(1)
rand(1)
```

The running command is:
```shell
mprof run --include-children python 
```

**Comment by ravinkohli** [comment]
I did not notice any significant changes with the memory footprint by making this change. Hence, we can close this
